satpy/tests/reader_tests/test_geos_area.py

```diff
@@ -0,0 +1,108 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+# Copyright (c) 2019 Satpy developers
+#
+# This file is part of satpy.
+#
+# satpy is free software: you can redistribute it and/or modify it under the
+# terms of the GNU General Public License as published by the Free Software
+# Foundation, either version 3 of the License, or (at your option) any later
+# version.
+#
+# satpy is distributed in the hope that it will be useful, but WITHOUT ANY
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR
+# A PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along with
+# satpy.  If not, see <http://www.gnu.org/licenses/>.
+"""Geostationary project utility module tests package.
+"""
+
+import sys
+
+from satpy.readers.geos_area import (get_xy_from_linecol,
+                                     get_area_extent,
+                                     get_area_definition)
+
+
+if sys.version_info < (2, 7):
+    import unittest2 as unittest
+else:
+    import unittest
+
+
+class TestGEOSProjectionUtil(unittest.TestCase):
+    """Tests for the area utilities"""
+
+    # Create dict with test data
+    pdict = {'a': 6378169.00,
+             'b': 6356583.80,
+             'h': 35785831.00,
+             'ssp_lon': 0.0,
+             'nlines': 3712,
+             'ncols': 3712,
+             'a_name': 'geostest',
+             'a_desc': 'test area',
+             'p_id': 'test_area',
+             'cfac': -13642337,
+             'lfac': -13642337,
+             'coff': 1856}
+
+    good_ext_s = (5567248.28340708,
+                  5567248.28340708,
+                  -5570248.686685662,
+                  -5570248.686685662)
+
+    good_ext = (5567248.28340708,
+                5570248.686685662,
+                -5570248.686685662,
+                -5567248.28340708)
+
+    def test_geos_area(self):
```

Comment:
Currently these are only full disk tests. Could you please add a test for segmented data as well? For example this is what you have in SEVIRI HRIT segments:

```
nlines, ncols, loff, coff, lfac, cfac, area_extent
464 3712 1856.0 1856.0 -13642337 -13642337 (5567248.28340708, -4175061.162145487, -5570248.686685662, -5567248.28340708)
464 3712 1392.0 1856.0 -13642337 -13642337 (5567248.28340708, -2782874.0408838945, -5570248.686685662, -4175061.162145487)
464 3712 928.0 1856.0 -13642337 -13642337 (5567248.28340708, -1390686.9196223018, -5570248.686685662, -2782874.0408838945)
464 3712 464.0 1856.0 -13642337 -13642337 (5567248.28340708, 1500.2016392905093, -5570248.686685662, -1390686.9196223018)
464 3712 0.0 1856.0 -13642337 -13642337 (5567248.28340708, 1393687.3229008832, -5570248.686685662, 1500.2016392905093)
464 3712 -464.0 1856.0 -13642337 -13642337 (5567248.28340708, 2785874.4441624754, -5570248.686685662, 1393687.3229008832)
464 3712 -928.0 1856.0 -13642337 -13642337 (5567248.28340708, 4178061.565424069, -5570248.686685662, 2785874.4441624754)
464 3712 -1392.0 1856.0 -13642337 -13642337 (5567248.28340708, 5570248.686685662, -5570248.686685662, 4178061.565424069)
```

Reply:
I've added a couple of segment tests now.